overlord/snapstate/snapstate_test.go

```diff
@@ -1871,6 +1872,172 @@ func (s *snapmgrTestSuite) TestInstallRunThrough(c *C) {
 	c.Assert(snapst.Required, Equals, false)
 }
 
+func (s *snapmgrTestSuite) TestParallelInstanceInstallRunThrough(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	tr := config.NewTransaction(s.state)
+	tr.Set("core", "experimental.parallel-instances", true)
+	tr.Commit()
+
+	chg := s.state.NewChange("install", "install a snap")
+	ts, err := snapstate.Install(s.state, "some-snap_instance", "some-channel", snap.R(0), s.user.ID, snapstate.Flags{})
+	c.Assert(err, IsNil)
+	chg.AddAll(ts)
+
+	s.state.Unlock()
+	s.settle(c)
+	s.state.Lock()
+
+	// ensure all our tasks ran
+	c.Assert(chg.Err(), IsNil)
+	c.Assert(chg.IsReady(), Equals, true)
+	c.Check(snapstate.Installing(s.state), Equals, false)
+	c.Check(s.fakeStore.downloads, DeepEquals, []fakeDownload{{
+		macaroon: s.user.StoreMacaroon,
+		// TODO parallel-install: fix once store changes are in place
+		name: "some-snap_instance",
+	}})
+	expected := fakeOps{
+		{
+			op:     "storesvc-snap-action",
+			userID: 1,
+		},
+		{
+			op: "storesvc-snap-action:action",
+			action: store.SnapAction{
+				Action:       "install",
+				InstanceName: "some-snap_instance",
+				Channel:      "some-channel",
+			},
+			revno:  snap.R(11),
+			userID: 1,
+		},
+		{
+			op:   "storesvc-download",
+			name: "some-snap_instance",
```

Comment:
this looks like a real bug or test confusion,  download shouldn't receive an instance name. I don't think we changed store to support that.

Reply:
either way, we probably should add a check about that in the fake download code

Reply:
I rechecked download uses the name only for errors afaict,  on the other end  some of those would be strange with the instance name, like the ones about paid snaps (you pay once, not per instance), something to think about

Reply:
I'll tweak the code to use snap name instead. Also will record target path in the tests.

Reply:
as I wrote,  the question is whether to pass instance-name or snap-name to Download, it matters mostly for error messages but some might be confusing with instance-name,  and some might be better with, I don't know if it means we should teach it about instance-name and splitting. Easiest for a start would be to pass snap-name